harvester_e2e_tests/integration/test_vm_functions.py

```diff
@@ -0,0 +1,374 @@
+
+        assert new_t > old_t, (
+            "Agent's probe time is not updated.\t"
+            f"Before softreboot: {old_t}, After softreboot: {new_t}\n"
+            f"Last API Status({code}): {data}"
+        )
+
+    def test_migrate(self, api_client, unique_vm_name, wait_timeout):
+        code, host_data = api_client.hosts.get()
+        assert 200 == code, (code, host_data)
+        code, data = api_client.vms.get_status(unique_vm_name)
+        cur_host = data['status']['nodeName']
+
+        new_host = next(h['id'] for h in host_data['data'] if cur_host != h['id'])
+
+        code, data = api_client.vms.migrate(unique_vm_name, new_host)
+        assert 204 == code, (code, data)
+
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        while endtime > datetime.now():
+            code, data = api_client.vms.get_status(unique_vm_name)
+            migrating = data['metadata']['annotations'].get("harvesterhci.io/migrationState")
+            if not migrating and new_host == data['status']['nodeName']:
+                break
+            sleep(5)
+        else:
+            raise AssertionError(
+                f"Failed to Migrate VM({unique_vm_name}) from {cur_host} to {new_host}\n"
+                f"API Status({code}): {data}"
+            )
+
+    def test_abort_migrate(self, api_client, unique_vm_name, wait_timeout):
+        code, host_data = api_client.hosts.get()
+        assert 200 == code, (code, host_data)
+        code, data = api_client.vms.get_status(unique_vm_name)
+        cur_host = data['status']['nodeName']
+
+        new_host = next(h['id'] for h in host_data['data'] if cur_host != h['id'])
+
+        code, data = api_client.vms.migrate(unique_vm_name, new_host)
+        assert 204 == code, (code, data)
+
+        states = ["Aborting migration", "Migrating"]
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        while endtime > datetime.now():
+            code, data = api_client.vms.get_status(unique_vm_name)
+            m_state = data['metadata']['annotations'].get("harvesterhci.io/migrationState")
+            if m_state == states[-1]:
+                states.pop()
+                if states:
+                    code, err = api_client.vms.abort_migrate(unique_vm_name)
+                    assert 204 == code, (code, err)
+                else:
+                    break
+            sleep(3)
+        else:
+            raise AssertionError(
+                f"Failed to abort VM({unique_vm_name})'s migration, stuck on {states[-1]}\n"
+                f"API Status({code}): {data}"
+            )
+
+        assert cur_host == data['status']['nodeName'], (
+            f"Failed to abort VM({unique_vm_name})'s migration,"
+            f"VM been moved to {data['status']['nodeName']} is not the origin host {cur_host}\n"
+        )
+
+    def test_delete(self, api_client, unique_vm_name, wait_timeout):
```

Comment:
Do we need to add the `--do-not-cleanup` check, or are we not doing that anymore?

Reply:
Consider following cases:
- A Test case failed, tear down steps (the _cleanup_) will fail too.
- A Test case passed, we will not look into the remaining environment, so _cleanup_ option is not necessary.

as above cases, we will not need the option anymore.

Reply:
I thought that the `--do-not-cleanup` was mostly for being able to get logs from the environment. I don't have much of an opinion either way I just wanted to check. If we are changing the policy on that though we should update the README.md and remove the requirement for `--do-not-cleanup` in new tests. CC @khushboo-rancher

Reply:
I do agree that it probably won't matter in a lot of the new tests. the `--do-not-cleanup` option was a bigger deal in the old involved complicated scenarios.

Reply:
Resolving due to not needing the `--do-not-cleanup` in this test.